### PR TITLE
fix: Replace 'Twitter' with 'X' to prevent 302/301 redirects

### PR DESCRIPTION
**Pull request by rajpatel2435**
Hi @joseluismolina @brandoncorbin @silent1mezzo 

### Description:

This pull request updates all instances of "Twitter" to "X" across the relevant components to prevent 302/301 redirects when integrating with the updated platform. The change ensures that references to the deprecated "Twitter" URL now point correctly to "X", eliminating any potential redirect issues.

I have tested this update in a new project setup, implementing the updated X component. The functionality works as expected without any redirects or errors, confirming the implementation is correct.

### Changes:
- Replaced all occurrences of "Twitter" with "X" in components and code files.
- Updated the `X` component to ensure proper integration without redirects.
- Updated related documentation (if applicable) to reflect the name change.

### Testing:
- Set up a new project with the updated code.
- Implemented the X component and verified that no 302/301 redirects occur.
- All functionality tested successfully in the new environment.

This update improves compatibility with the new platform and avoids the redirect errors caused by the outdated Twitter references.

### CODE REVIEW

**Comment by silent1mezzo** [comment]
This full change will require a major bump since it'd break anyone who's importing the components and customizing them. I'm more inclined to just change the URL and keep the component names the same to prevent this

**Comment by rajpatel2435** [comment]
Hi @silent1mezzo 

The repo has been updated back to twitter .

Could you please check this out?

Thanks

**Comment by silent1mezzo** [comment]
@rajpatel2435 can you revert the file change too? Then we can do that in a separate 3.x launch

**Comment by rajpatel2435** [comment]
Hi @silent1mezzo 

the package.json or all file changes?

File: src/X.astro
Comment (silent1mezzo) [inline src/X.astro:1]: Can this be reverted back to Twitter for this release?

Comment (silent1mezzo) [inline src/X.astro:1]: This should be reverted for this release.

File: tests/button.test.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import EmailShareButton from "../src/Email.astro";
 import FacebookShareButton from '../src/Facebook.astro';
 import HackerNewsShareButton from '../src/HackerNews.astro';
 import LinkedInShareButton from '../src/LinkedIn.astro';
-import TwitterShareButton from '../src/Twitter.astro';
+import TwitterShareButton from '../src/X.astro';
```
Comment (silent1mezzo) [inline tests/button.test.ts]: This should be reverted for this release.

File: index.ts [hunk 2]
```diff
@@ -3,6 +3,6 @@ export { default as FacebookShareButton } from './src/Facebook.astro';
 export { default as HackerNewsShareButton } from './src/HackerNews.astro';
 export { default as LinkedInShareButton } from './src/LinkedIn.astro';
 export { default as RedditShareButton } from './src/Reddit.astro';
-export { default as TwitterShareButton } from './src/Twitter.astro';
```
Comment (silent1mezzo) [inline index.ts]: This should be reverted for this release.

**Review by silent1mezzo — CHANGES_REQUESTED** [review]

File: index.ts (hunk 2 above)
Comment (rajpatel2435) [reply index.ts]: Hi @silent1mezzo 

Updated pull request

**Comment by rajpatel2435** [comment]
Hi @silent1mezzo 

Can you please check this out? As I removed the unnecessary stuff that your requested.

Thanks

**Comment by silent1mezzo** [comment]
Looks good, thanks @rajpatel2435. I'll cut a release later tonight

**Comment by silent1mezzo** [comment]
2.0.3 has been [published](https://www.npmjs.com/package/astro-social-share/v/2.0.3)

**Pull request merged**